valhalla/baldr/curl_tilegetter.h

```diff
@@ -1,11 +1,12 @@
 #pragma once
```

Comment:
what if we added an optional string parameter on the end of the get function that, if non empty, will be the place to store the result to. that way we can:

1. avoid having to write the code both in sample and in graphtile
2. we can move the specific filesystem stuff here that we need

Reply:
2. filesystem.h cleaned, all valhalla specific functions removed and only general filesystem functions left.

Reply:
Concerning 1, as I understood you meant to hide the logic of storing data to filesystem inside the get function (from tile_getter_t). At first the get function will retrieve data from remote storage and then inside the same get function ( in case we pass additional parameter (path where to store data locally after retrieving them from remote storage) to this method) we will store the retrieved data.
As for me this sounds like not so good idea 
1.  difficult to guess from the function name that this method not only retrieves data but also stores them.
2. Do we always need/can to store data when we pass this additional parameter or not? 
3. Logic of storing data and logic of retrieving data from remote storage looks completely separate/unrelated, as for me. It might be better just to reuse (any of the already implemented ) cache from (e.g.) GraphReader, combine it with the logic from tile_getter_t and create something like StorageManager. StorageManager will check the cache and use the get function if nothing was found in the cache. This will help to keep code more structured, reusable and understandable.